react-native/react/login2/register/set-public-name/index.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+/* @flow */
+
+import React, {Component} from '../../../base-react'
+import {setDeviceName} from '../../../actions/login2'
+import Render from './index.render'
+
+export default class SetPublicName extends Component {
+  nameTaken (deviceName) {
+    return this.props.existingDevices && this.props.existingDevices.indexOf(deviceName) !== -1
+  }
+
+  submitEnabled (deviceName) {
+    return deviceName && deviceName.length && !this.nameTaken(deviceName)
+  }
+
+  render () {
+    return (
+      <Render
+        onSubmit={ deviceName => this.props.dispatch(setDeviceName(deviceName)) }
```

Comment:
We want the smart component to hoist up this kind of logic so its not in the render methods